plugin/evm/atomic/vm/tx_gossip_test.go

```diff
@@ -0,0 +1,312 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package vm
+
+import (
+	"context"
+	"encoding/binary"
+	"sync"
+	"testing"
+	"time"
+
+	avalancheatomic "github.com/ava-labs/avalanchego/chains/atomic"
+	"github.com/ava-labs/avalanchego/database/memdb"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/network/p2p"
+	"github.com/ava-labs/avalanchego/network/p2p/gossip"
+	"github.com/ava-labs/avalanchego/proto/pb/sdk"
+	"github.com/ava-labs/avalanchego/snow"
+	"github.com/ava-labs/avalanchego/snow/engine/enginetest"
+	"github.com/ava-labs/avalanchego/snow/snowtest"
+	"github.com/ava-labs/avalanchego/snow/validators"
+	agoUtils "github.com/ava-labs/avalanchego/utils"
+	"github.com/ava-labs/avalanchego/utils/crypto/secp256k1"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/avalanchego/utils/set"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/vms/components/avax"
+	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/ava-labs/coreth/plugin/evm/atomic"
+	"github.com/ava-labs/coreth/plugin/evm/config"
+	"github.com/ava-labs/coreth/plugin/evm/vmtest"
+	"github.com/ava-labs/coreth/utils"
+)
+
+func TestAtomicTxGossip(t *testing.T) {
+	require := require.New(t)
+	ctx := context.Background()
+	snowCtx := snowtest.Context(t, snowtest.CChainID)
+	snowCtx.AVAXAssetID = ids.GenerateTestID()
+	validatorState := utils.NewTestValidatorState()
+	snowCtx.ValidatorState = validatorState
+	memory := avalancheatomic.NewMemory(memdb.New())
+	snowCtx.SharedMemory = memory.NewSharedMemory(snowCtx.ChainID)
+
+	pk, err := secp256k1.NewPrivateKey()
+	require.NoError(err)
+	address := pk.EthAddress()
+	genesis := vmtest.NewPrefundedGenesis(100_000_000_000_000_000, address)
+	genesisBytes, err := genesis.MarshalJSON()
+	require.NoError(err)
+
+	responseSender := &enginetest.SenderStub{
+		SentAppResponse: make(chan []byte, 1),
+	}
+	vm := newAtomicTestVM()
```

Comment:
As you introduced a new TestVM function setup, should the use of `newAtomicTestVM()` be integrated with those functions? (e.g. `setupDefaultTestVM`)

Reply:
I reverted that, I think it's better to utilize first create then setup in everywhere.